File: tabulate-core/src/test/java/io/github/voytech/tabulate/core/builder/fluent/support/TestOutputBinding.java
package io.github.voytech.tabulate.core.builder.fluent.support;

import io.github.voytech.tabulate.template.result.OutputBinding;
import kotlin.Unit;
import org.jetbrains.annotations.NotNull;

public class TestOutputBinding implements OutputBinding<TestRenderingContext, Unit> {
    @NotNull
    @Override
    public Class<Unit> outputClass() {
        return Unit.class;
    }

    @Override
    public void setOutput(@NotNull TestRenderingContext renderingContext, Unit output) {

    }

    @Override
    public void flush() {

    }
}

File: tabulate-core/src/test/kotlin/io/github/voytech/tabulate/support/TestOutputBinding.kt
package io.github.voytech.tabulate.support

import io.github.voytech.tabulate.core.reify
import io.github.voytech.tabulate.core.result.OutputBinding
import java.util.logging.Logger

class TestOutputBinding: OutputBinding<TestRenderingContext, Unit> {

    override fun outputClass(): Class<Unit> = reify()

    override fun flush() {
        logger.info("This is fake implementation of ResultProvider flushing results into ether")
    }

    override fun setOutput(renderingContext: TestRenderingContext, output: Unit) {
        logger.info("This is fake implementation of ResultProvider flushing results into ether")
    }

    companion object {
        val logger: Logger = Logger.getLogger(TestOutputBinding::class.java.name)
    }
}

